src/linux/BTSerialPortBindingServer.cc

```diff
@@ -306,10 +288,13 @@ void BTSerialPortBindingServer::EIO_AfterRead(uv_work_t *req) {
         sprintf(msg, "Error reading from connection: errno: %d", baton->errorno);
         argv[0] = Nan::Error(msg);
         argv[1] = Nan::Undefined();
-        if(baton->errorno == ECONNRESET || baton->size == 0){
+        if(baton->errorno == ECONNRESET || baton->errorno == ETIMEDOUT || baton->size == 0 || baton->isDisconnect){
+            baton->rfcomm->CloseClientSocket();
+            if (!baton->isClose) {
```

Comment:
why would this happen?

Reply:
ETIMEDOUT is what I got when I either walked out of range with the scanner or did a battery pull.  I'm not sure when ECONNRESET gets triggered, but I left it in there since it was in the original code.